setup.sh

```diff
@@ -12,8 +16,88 @@ _addbin() {
     [ ! -z "${1}" ] && export PATH="${1}:${PATH}"
 }
 
+parse_arguments() {
+    # Default values
+    DEFAULT_ANALYSIS="KingMaker"
+    DEFAULT_ENV_PATH=""
+    DEFAULT_LIST_WORKFLOWS=false
+    DEFAULT_CROWN_ANALYSIS=""
+    ANALYSIS=${DEFAULT_ANALYSIS}
+    ENV_PATH=${DEFAULT_ENV_PATH}
+    LIST_WORKFLOWS=${DEFAULT_LIST_WORKFLOWS}
+    CROWN_ANALYSIS=${DEFAULT_CROWN_ANALYSIS}
+
+    # Parse arguments
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+            -a|--analysis)
+                ANALYSIS="$2"
+                shift 2
+                ;;
+            -e|--env-path)
+                ENV_PATH="$2"
+                shift 2
+                ;;
+            -c|--crown-analysis)
+                CROWN_ANALYSIS="$2"
+                shift 2
+                ;;
+            -l|--list)
+                echo "Available workflows:"
+                echo "-------------------"
+                for workflow in "${ANA_LIST[@]}"; do
+                    if [[ "${workflow}" == "${DEFAULT_ANALYSIS}" ]]; then
+                        echo "* ${workflow} (default)"
+                    else
+                        echo "* ${workflow}"
+                    fi
+                done
+                return 1
+                ;;
+            -h|--help)
+                echo "Usage: source setup.sh [options]"
+                echo ""
+                echo "Options:"
+                echo "  -a, --analysis ANALYSIS    Specify the analysis workflow to use"
+                echo "                            [default: ${DEFAULT_ANALYSIS}]"
+                echo "  -e, --env-path PATH       Specify custom environment path"
+                echo "                            [default: auto-detected]"
+                echo "  -c, --crown-analysis NAME  Specify CROWN analysis to check out"
+                echo "                            (only with KingMaker workflow)"
+                echo "                            Available: tau, earlyrun3, whtautau,"
+                echo "                            boosted_h_tautau, s"
```

Comment:
Can we just add a link to the documentation? By doing that we don't have to care to change it here if new analyses are added.
https://crown.readthedocs.io/en/latest/introduction.html#id1